kms/softkms/softkms.go

```diff
@@ -73,8 +73,7 @@ func (k *SoftKMS) CreateSigner(req *apiv1.CreateSignerRequest) (crypto.Signer, e
 	var opts []pemutil.Options
 	if req.Password != nil {
 		opts = append(opts, pemutil.WithPassword(req.Password))
-	}
-	if req.Password == nil && req.PasswordPrompter != nil {
+	} else if req.Password == nil && req.PasswordPrompter != nil {
```

Comment:
The else condition is Passord == nil 

```suggestion
	} else if req.PasswordPrompter != nil {
```